### PR TITLE
Batch decoding python bypass #315

### PR DESCRIPTION
**Pull request by chirantan31**

### CODE REVIEW

File: honeybadgermpc/ntl/rsdecode.pxd
Comment (Drake-Eidukas) [inline honeybadgermpc/ntl/rsdecode.pxd]: Looks like you have some merge conflicts here btw

**Review by Drake-Eidukas — CHANGES_REQUESTED** [review]
Just a few questions--
- What is the new library and why do we need it?
  - On a side note, we'll probably want to make this a git submodule. 
  - What sort of improvements does this provide?
 
Other than that I'm not too well versed in those parts of the codebase, so I'll leave reviewing that up to @amiller .

Some things to do before merging this:

- [x] Squash this down to one commit using `git rebase -i dev`
  - Make sure that you pulled in the newest version of dev
- [x] Remove remaining merge conflict fragments
- [x] Add unit tests for the new functionality (@amiller can probably give insight on that, not sure how it's done in this part of the codebase if at all)
- [x] Add a concisely headed but descriptive commit message summarizing the changes here.

**Comment by Drake-Eidukas** [comment]
As for getting the tests to work on travis, this should be working very soon. In the meantime, work on those tasks.

**Comment by rahulgovind** [comment]
Quickly went through the code and just had a few high-level comments/questions

- You are using variable-length encoding here where you first encode the number of limbs `s`, and then you serialize `s` limbs (which are unsigned longs?) after that. However, since the intermediate values are uniformly distributed, they will mostly occupy as many bytes as `p`. Therefore, It is not clear to me why this variable-length encoding would be better than a fixed-length encoding where you just transmit 32 bytes per ZZ_p

- It looks like you are assuming you will receive valid serialized data from the other parties which might not be the case. For example, a large value of `size` would just crash your node.  I would suggest adding additional checks while deserializing wherever possible.

**Comment by chirantan31** [comment]
Closed in lieu of #353

**Comment by amiller** [comment]
> Just a few questions--
> 
> * What is the new library and why do we need it?
>   
>   * On a side note, we'll probably want to make this a git submodule.
>   * What sort of improvements does this provide?
> 
> Other than that I'm not too well versed in those parts of the codebase, so I'll leave reviewing that up to @amiller .
> 
> Some things to do before merging this:
> 
> * [x]  Squash this down to one commit using `git rebase -i dev`
>   
>   * Make sure that you pulled in the newest version of dev
> * [x]  Remove remaining merge conflict fragments
> * [x]  Add unit tests for the new functionality (@amiller can probably give insight on that, not sure how it's done in this part of the codebase if at all)
> * [x]  Add a concisely headed but descriptive commit message summarizing the changes here.

I think the existing unit tests actually cover all of this. It is not actually new functionality, rather it is a different method (avoids serialization) for the existing functionality.

File: honeybadgermpc/ntl/rsdecode_impl.h
Comment (rahulgovind) [inline honeybadgermpc/ntl/rsdecode_impl.h:459]: There should be an extra check for size here (size <= 32?). Also, is there a way to check if HPS serialization failed for any reason?

**Review by rahulgovind — COMMENTED** [review]
The issues in https://github.com/initc3/HoneyBadgerMPC/pull/352#issuecomment-511182296 might still be relevant here.

File: honeybadgermpc/ntl/rsdecode_impl.h
Comment (rahulgovind) [reply honeybadgermpc/ntl/rsdecode_impl.h:459]: Meant to review #353. You can ignore this.

Comment (amiller) [reply honeybadgermpc/ntl/rsdecode_impl.h:459]: great point, i'd want to be able to catch (in python) any exception due to serialization failure

**Pull request closed without merging**
